packages/frontend/src/components/SingleSelect/types.ts

```diff
@@ -14,6 +14,8 @@ export type ComboboxItem<T = string> =
       icon?: As
       /** Badge to display on the right of the label, if available */
       badge?: JSX.Element
+      /** Whether this is an option to add new */
+      isAddNew?: boolean
```

Comment:
nit: wondering if it make sense to make this more generic, i.e. allow specifying onSelect callbacks on any item. so you have something like
* Extend `ComboBoxItem` to have a `callback` attribute
* define `hasOnSelectCallback` instead of `isItemAddNew` etc

But not a priority for now.

Reply:
I just working through the `filteredString` stuff. Ok maybe a bit tricky to make this generic :/

Reply:
true that but isAddNew items do not set the value. whereas on select callbacks can just depend on a useEffect when the value is changed